### PR TITLE
Add sentry fingerprint

### PR DESCRIPTION
**Pull request by kreek**
## Description of change
Currently we have many error keys (e.g. SM115, VET360_EMAIL300) for `Common::Exceptions::BackendServiceExceptions` but they end up being grouped together in Sentry. This PR uses the key as the sentry fingerprint so that each key will get its own grouping.

## Testing done
Tested locally against a test sentry account.

## Testing planned
Staging integration tests.

#### Unique to this PR
- [ ] `Common::Exceptions::BackendServiceExceptions` are no longer grouped

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: config/initializers/raven.rb [hunk 1]
```diff
@@ -11,7 +11,13 @@
...
   config.excluded_exceptions += ['Sentry::IgnoredError']
 
-  config.async = lambda { |event|
-    SentryJob.perform_async(event)
-  }
+  config.before_send = lambda do |event, hint|
+    return event unless hint[:exception] && hint[:exception].is_a?(Common::Exceptions::BackendServiceException)
+
+    event.fingerprint = hint[:exception].key
+    event.tags = event.tags.merge!(key: hint[:exception].key)
```
Comment (lihanli) [inline config/initializers/raven.rb]: `event.tags =` shouldn't be needed if you're using `merge!`

**Comment by annaswims** [comment]
The sentry client version is ahead of the sentry server version and we haven't seen any errors in sentry since this was deployed. Can we roll this back?

**Comment by saneshark** [comment]
This is my fault. The raven-ruby gem introduced fingerprinting, but that has been a part of the Raven API long before raven-ruby got around to introducing it as a feature. 

Do we need to look at upgrading sentry server using master?

**Comment by kreek** [comment]
@saneshark I would say it was my fault since I pushed the button :) I'd like to try to repro this locally. From the web UI it looks like we're running the 9.0.0 tag? (not 9.0.0rc1)

**Comment by kreek** [comment]
... nevermind catching up with email and I see the comments about the last compatible gem version here https://github.com/department-of-veterans-affairs/vets-api/pull/2806

**Pull request merged**